### PR TITLE
Add SQL Backup Retention fields

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/7961

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
sql: added `settings.0.backup_configuration.transaction_log_retention_days` and `settings.0.backup_configuration.transaction_log_retention_days` fields to `google_sql_database_instance`
```

### CODE REVIEW

File: third_party/terraform/tests/resource_sql_database_instance_test.go.erb [hunk 1]
```diff
@@ -1219,3 +1242,26 @@ resource "google_sql_database_instance" "instance" {
...
+  settings {
+    tier = "db-f1-micro"
+    backup_configuration {
+      enabled                        = true
+      start_time                     = "00:00"
+      binary_log_enabled             = true
+	  transaction_log_retention_days = 2
+	  backup_retention_settings {
+	    retained_backups = 4
+	  }
```
Comment (ScottSuarez) [inline third_party/terraform/tests/resource_sql_database_instance_test.go.erb:1262]: tabs vs spaces

**Review by ScottSuarez — APPROVED** [review]

File: third_party/terraform/tests/resource_sql_database_instance_test.go.erb (hunk 1 above)
Comment (upodroid) [reply third_party/terraform/tests/resource_sql_database_instance_test.go.erb]: fixed

**Comment by ScottSuarez** [comment]
```
go: downloading github.com/dnaeon/go-vcr v1.0.1
go: extracting github.com/dnaeon/go-vcr v1.0.1
go: finding github.com/dnaeon/go-vcr v1.0.1
# github.com/hashicorp/terraform-provider-google/v3/google
google/resource_sql_database_instance_test.go:1267:6: expected '(', found testAccSqlDatabaseInstance_beforeBackup
google/resource_sql_database_instance_test.go:1286:6: expected '(', found testAccSqlDatabaseInstance_restoreFromBackup
google/resource_sql_database_instance_test.go:1313:3: expected '}', found 'EOF'
FAIL	github.com/hashicorp/terraform-provider-google/v3/google [setup failed]
?   	github.com/hashicorp/terraform-provider-google/v3	[no test files]
?   	github.com/hashicorp/terraform-provider-google/v3/scripts/affectedtests	[no test files]
?   	github.com/hashicorp/terraform-provider-google/v3/scripts/sidebar	[no test files]
?   	github.com/hashicorp/terraform-provider-google/v3/version	[no test files]
FAIL
GNUmakefile:11: recipe for target 'test' failed
make: *** [test] Error 1
```

**Comment by upodroid** [comment]
There is something weird with the code that is conflicting.

**Comment by upodroid** [comment]
It is fixed now.

**Comment by ScottSuarez** [comment]
Screen Shot 2021-01-15 at 11 36 30 AM

**Comment by upodroid** [comment]
I ran the `TestAccSqlDatabaseInstance_basic_with_user_labels` and no errors we reported.

Have a look at this gist. Debugging was enabled.

https://gist.github.com/upodroid/a0474a964a99f08e2d7347938b1418e3

**Comment by upodroid** [comment]
On a sidenote, SQL Admin is one of the most dodgiest Cloud APIs from experience.

**Comment by upodroid** [comment]
Fixed conflicts

**Comment by upodroid** [comment]
Bump

**Comment by ScottSuarez** [comment]
/gcbrun - fixing my tracking of issues like this so I don't lose them. I tended to use the issues tab on the top which actually makes me lose track of issues one I do an initial pass. Thanks for your patience upodroid

**Pull request merged**